src/pybind/mgr/dashboard/frontend/src/app/ceph/pool/pool-form/pool-form.component.ts

```diff
@@ -0,0 +1,530 @@
+  }
+
+  private pgCalc() {
+    const poolType = this.form.getValue('poolType');
+    if (!this.info || this.form.get('pgNum').dirty || !poolType) {
+      return;
+    }
+    const pgMax = this.info.osd_count * 100;
+    const pgs =
+      poolType === 'replicated' ? this.replicatedPgCalc(pgMax) : this.erasurePgCalc(pgMax);
+    if (!pgs) {
+      return;
+    }
+    const oldValue = this.data.pgs;
+    this.pgUpdate(pgs);
+    const newValue = this.data.pgs;
+    if (!this.externalPgChange) {
+      this.externalPgChange = oldValue !== newValue;
+    }
+  }
+
+  private replicatedPgCalc(pgs): number {
+    const sizeControl = this.form.get('size');
+    const size = sizeControl.value;
+    if (sizeControl.valid && size > 0) {
+      return pgs / size;
+    }
+  }
+
+  private erasurePgCalc(pgs): number {
+    const ecpControl = this.form.get('erasureProfile');
+    const ecp = ecpControl.value;
+    if ((ecpControl.valid || ecpControl.disabled) && ecp) {
+      return pgs / (ecp.k + ecp.m);
+    }
+  }
+
+  private pgUpdate(pgs?, jump?) {
+    pgs = _.isNumber(pgs) ? pgs : this.form.getValue('pgNum');
+    if (pgs < 1) {
+      pgs = 1;
+    }
+    let power = Math.round(Math.log(pgs) / Math.log(2));
+    if (_.isNumber(jump)) {
+      power += jump;
+    }
+    if (power < 0) {
+      power = 0;
+    }
+    pgs = Math.pow(2, power); // Set size the nearest accurate size.
+    this.data.pgs = pgs;
+    this.form.silentSet('pgNum', pgs);
+  }
+
+  private setComplexValidators() {
+    if (this.editing) {
+      this.form
+        .get('pgNum')
+        .setValidators(
+          CdValidators.custom('noDecrease', (pgs) => this.data.pool && pgs < this.data.pool.pg_num)
+        );
+    } else {
+      CdValidators.validateIf(
+        this.form.get('size'),
+        () => this.form.get('poolType').value === 'replicated',
+        [
+          CdValidators.custom(
+            'min',
+            (value) => this.form.getValue('size') && value < this.getMinSize()
+          ),
+          CdValidators.custom(
+            'max',
+            (value) => this.form.getValue('size') && this.getMaxSize() < value
+          )
+        ]
+      );
+    }
+    this.setCompressionValidators();
+  }
+
+  private setCompressionValidators() {
+    CdValidators.validateIf(this.form.get('minBlobSize'), () => this.activatedCompression(), [
+      Validators.min(0),
+      CdValidators.custom('maximum', (size) =>
+        this.compareBlobSize(size, this.form.getValue('maxBlobSize'))
+      )
+    ]);
+    CdValidators.validateIf(this.form.get('maxBlobSize'), () => this.activatedCompression(), [
+      Validators.min(0),
+      CdValidators.custom('minimum', (size) =>
+        this.compareBlobSize(this.form.getValue('minBlobSize'), size)
+      )
+    ]);
+    CdValidators.validateIf(this.form.get('ratio'), () => this.activatedCompression(), [
+      Validators.min(0),
+      Validators.max(1)
+    ]);
+  }
+
+  private compareBlobSize(minimum, maximum) {
```

Comment:
Better use this code because maximum may contain the string '-' which results in a comparision of 123456 >= null.

```
  private compareBlobSize(minimum, maximum) {
    minimum = this.formatter.toBytes(minimum);
    maximum = this.formatter.toBytes(maximum);
    if (_.isNull(minimum) || _.isNull(maximum)) {
      return false;
    }
    return minimum >= maximum;
  }
```